src/notebook/Events.jl

```diff
@@ -83,15 +84,21 @@ FileEditEvent(notebook::Notebook) = begin
     FileEditEvent(notebook, file_contents, notebook.path)
 end
 
-# Triggered when we open a new notebook
+# Triggered when we create a new notebook
 struct NewNotebookEvent <: PlutoEvent
 end
 
-# Triggered when a user opens a notebook
+# Triggered when we open any notebook
 struct OpenNotebookEvent <: PlutoEvent
     notebook::Notebook
 end
 
+# Triggered when Pluto completes an evaluation loop
+struct NotebookExecutionDoneEvent <: PlutoEvent
+    notebook::Notebook
+    user_requested_run::Bool
```

Comment:
do you also want to fire this event for bond updates? maybe add a field  `from_bond_update::Bool`

Reply:
It already runs on bond updates! `user_requested_run::Bool` is a bond update if I'm not mistaken.